config/src/main/java/com/typesafe/config/impl/BeanFactory.java

```diff
@@ -0,0 +1,150 @@
+package com.typesafe.config.impl;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sent as pull request to config project.
+ * https://github.com/typesafehub/config/pull/107
+ */
+public class BeanFactory {
+
+    public static <T> T create(Config config, Class<T> clazz) {
+        return createInternal(config, clazz);
+    }
+
+    private static final Map<TimeUnit,List<String>> TIME_UNITS = new HashMap<>();
+    static {
+        TIME_UNITS.put(TimeUnit.NANOSECONDS, Arrays.asList("ns", "nano", "nanos", "nanosecond", "nanoseconds"));
```

Comment:
Are these all valid? It doesn't look like what is being used elsewhere in the code. See:
https://github.com/typesafehub/config/blob/master/config/src/main/java/com/typesafe/config/impl/SimpleConfig.java#L522